src/test/java/com/pi4j/crowpi/components/SoundSensorComponentTest.java

```diff
@@ -0,0 +1,61 @@
+package com.pi4j.crowpi.components;
+
+import com.pi4j.crowpi.ComponentTest;
+import com.pi4j.io.gpio.digital.DigitalInput;
+import com.pi4j.io.gpio.digital.DigitalState;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SoundSensorComponentTest extends ComponentTest {
+    protected SoundSensorComponent soundSensor;
+    protected DigitalInput din;
+
+    @BeforeEach
+    public void setUp() {
+        soundSensor = new SoundSensorComponent(pi4j);
+        din = soundSensor.getDigitalInput();
```

Comment:
Here you would use `toMock(soundSensor.getDigitalInput())` instead, which will transform the generic `DigitalInput` into its specific type `MockDigitalInput`.